src/typ.rs

```diff
@@ -1428,9 +1428,9 @@ pub fn infer_module(
                         typ: typ.clone(),
                         origin: location.clone(),
                         variant: ValueConstructorVariant::ModuleFn {
-                            name: name.clone(),
+                            name: fun.clone(),
                             field_map: field_map.clone(),
-                            module: module_name.clone(),
+                            module: vec![module.clone()],
```

Comment:
Wow that was a lot simpler than I expected!

Reply:
Haha yeah, me too. I actually implemented it in a much more complicated way first and then went "wait, what if I just..."